### PR TITLE
Add protocol tests

### PR DESCRIPTION
**Pull request by UserNotFound**
Also run tests after build is completed.

### CODE REVIEW

File: test/ssl-test.bats [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bats
+
+source "${BATS_TEST_DIRNAME}/test_helpers.sh"
+
+local_s_client() {
+  echo OK | openssl s_client -connect localhost:5671 "$@"
```
Comment (krallin) [inline test/ssl-test.bats]: Can you test port 15671 (the management daemon) as well?

File: test/ssl-test.bats [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+@test "It should allow connections using TLS1.2" {
+  start_rabbitmq
+
+  run local_s_client -tls1_2
+  [ "$status" -eq 0 ]
```
Comment (krallin) [inline test/ssl-test.bats]: Same comment as in https://github.com/aptible/docker-influxdb/pull/6 regarding the un-necessary use of `run` here

File: Dockerfile.erb [hunk 3]
```diff
@@ -51,7 +51,6 @@ RUN         ln -s /var/db/testca/cacert.pem /ssl/cacert.pem && \
             ln -s /var/db/server/key.pem /ssl/key.pem
 
 ADD         test /tmp/test
-RUN         bats --tap /tmp/test && apk del --purge python
```
Comment (krallin) [inline Dockerfile.erb]: Thanks for breaking this out :+1:

File: test/ssl-test.bats (hunk 1 above)
Comment (UserNotFound) [reply test/ssl-test.bats]: Ah, good catch!

**Pull request merged**